python/pubmed_crawler.py

```diff
@@ -0,0 +1,343 @@
+                center_id = ", ".join(list(set(center.id)))
+                center_name = ", ".join(list(set(center.consortium)))
+            consortium_grants = ", ".join(
+                [center.grantType.iloc[0] + " " + grant for grant in grants])
+
+            # KEYWORDS
+            keywords = soup.find(attrs={"class": "keywords"})
+            keywords = keywords.find("p").text if keywords else ""
+
+            # RELATED INFORMATION
+            related_info = get_related_info(pmid)
+
+            #   GEO
+            gse_ids = parse_geo(related_info.get('gds'))
+            gse_url = make_urls(
+                "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=", gse_ids)
+
+            #   SRA
+            srx, srp = parse_sra(related_info.get('sra'))
+            srx_url = make_urls("https://www.ncbi.nlm.nih.gov/sra/", srx)
+            srp_url = make_urls(
+                "https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=", srp)
+
+            #   DBGAP
+            dbgaps = parse_dbgap(related_info.get('gap'))
+            dbgap_url = make_urls(
+                "https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=", dbgaps)
+
+            row = pd.DataFrame(
+                [[center_id, center_name, url, journal, year, title, authors,
+                  consortium_grants, ", ".join(srx), srx_url, ", ".join(srp),
+                  srp_url, ", ".join(dbgaps), dbgap_url, gse_url,
+                  "", keywords]],
+                columns=columns)
+            table.append(row)
+            session.close()
+
+            # Increment progress bar animation.
+            progress()
+
+    return pd.concat(table)
+
+
+def find_publications(syn, args):
+    """Get list of publications based on grants of consortia."""
+
+    consortium = get_view(syn, args.grantview_id)
+    grants = get_grants(consortium)
+    pmids = get_pmids(grants)
+
+    # If user provided a table ID, only scrap info from
+    # publications not already listed in the provided table.
+    if args.table_id:
+        print(f"Comparing with table {args.table_id}...")
+        current_publications = syn.tableQuery(
+            f"select * from {args.table_id}").asDataFrame()
+        current_pmids = {re.search(r"[/=](\d+)$", i).group(1)
+                         for i in list(current_publications.PubMed)}
+        pmids -= current_pmids
+        print(f"  New publications found: {len(pmids)}\n")
+
+    print(f"Pulling information from publications...")
+
+    table = scrap_info(pmids, grants, consortium, args.consortium_name)
+    table.to_csv(args.output + ".csv", index=False, sep="\t", encoding="utf-8")
+    print("DONE")
+
+
+def main():
+    """Main function."""
+
+    syn = login()
+    args = get_args()
+
+    # In order to make >3 Entrez requests/sec, 'email' and 'api_key'
+    # params need to be set. Using Milen's info...
+    Entrez.email = 'milen.nikolov@sagebase.org'
+    Entrez.api_key = '3f8cfef8d4356963e36d145c96b9ca9ece09'
+
```

Comment:
Let's use environment variables for credentials, just to be safe. Here's an example:
https://github.com/Sage-Bionetworks/prov-service/blob/develop/synprov/config.py#L25-L26